<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="robot_name" default="dingo2"/>
    <arg name="obstacle_name1" default="dynamic_object3"/>
    <arg name="rviz" default="true"/>
    <arg name="start_delay_seconds" default="0"/>
    <rosparam command="load" file="$(find fabrics_bridge)/config/fabrics_kinova_config.yaml"/>
<!-- 
    <group ns="fabrics"> -->
        <!-- start fabric node -->
    <node pkg="fabrics_bridge" type="kinova_fabrics_node" name="kinova_fabrics_node" output="screen"/>

    <!-- <node pkg="fabrics_bridge" type="kinova_kortex_node" name="kinova_kortex_node" output="screen" launch-prefix="bash -c 'sleep $(arg start_delay_seconds); $0 $@' ">
        <param name="robot_name" value="$(arg robot_name)"/>
    </node> -->
    <!-- </group> -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find fabrics_bridge)/config/fabrics_boxer_visualization.rviz" if="$(arg rviz)"/> -->
    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dingo_planners)/rviz/demo.rviz"/> -->

    <include file="$(find vicon_bridge)/launch/vicon.launch">
        <arg name="only_use_object_specific" default="true"/>
        <arg name="object_names" value="[$(arg robot_name), $(arg obstacle_name1)]"/>
        <arg name="object_msg_types" value="[geometry_msgs/PoseWithCovarianceStamped, geometry_msgs/PoseWithCovarianceStamped]"/>
        <arg name="object_frame_ids" value="[map, map]"/>
        <arg name="object_publish_topics" value="[/vicon/$(arg robot_name), /vicon/obstacle1]"/>
        <arg name="object_frequency_divider" value="[1, 1]"/>
    </include>

    <!-- Augment the pose readings with velocity estimates via an EKF -->
    <node pkg="robot_localization" type="ekf_localization_node" name="robot_ekf" clear_params="true">
        <rosparam command="load" file="$(find dingo_planners)/config/vicon_ekf.yaml"/>
        <remap from="pose" to="/vicon/$(arg robot_name)" />
        <remap from="/odometry/filtered" to="/robot_ekf/odometry"/>
    </node>

    <!-- Basic visualization -->
    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dingo_planners)/rviz/demo.rviz"/> -->
</launch>
